### PR TITLE
Support for OpenTelemetry (real-time logging of events and errors)

### PR DESCRIPTION
**Pull request by h-arlt**
Adds basic support for OpenTelemetry and use it for real-time logging of errors and events.

For the rationale behind this PR see #514 .

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/engine/metrics/otel/OtelMetricsReporter.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    private void reportTransaction(final TransactionData txnData)
+    {
+        // Only process failed transactions
+        if (txnData.hasFailed())
+        {
+            logRecord((record) -> {
+                record.setObservedTimestamp(Instant.now());
+                record.setTimestamp(Instant.ofEpochMilli(txnData.getTime()));
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/engine/metrics/otel/OtelMetricsReporter.java:84]: Better use the endTime value, as this is usually much closer to the time when the log record occurred?

```suggestion
                record.setTimestamp(Instant.ofEpochMilli(txnData.getEndTime()));
```

**Review by jowerner — CHANGES_REQUESTED** [review]

File: src/main/java/com/xceptance/xlt/engine/metrics/otel/OpenTelemetryConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.xceptance.xlt.api.util.XltProperties;
+import com.xceptance.xlt.common.XltConstants;
+
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+
+public final class OpenTelemetryConfiguration
```
[the remark this answers is not in the document]
Comment (h-arlt) [reply src/main/java/com/xceptance/xlt/engine/metrics/otel/OpenTelemetryConfiguration.java:15]: I've oriented on the official OTel Java examples that always name OTel initialization classes with a suffix in `Configuration`.  Maby, the rationale behind this is that an application should not use multiple OpenTelemetrySdk instances but only one and that this completes contains all code to configure the (single) OpenTelemetrySdk instance.

On the other side, a factory acts as object construction facility whereas the objects are often used for a concrete use-case only and destroyed afterwards (short-living, non-singletons).

File: src/main/java/com/xceptance/xlt/engine/metrics/otel/OtelMetricsReporter.java (hunk 1 above)
Comment (h-arlt) [reply src/main/java/com/xceptance/xlt/engine/metrics/otel/OtelMetricsReporter.java]: Done.

File: src/main/java/com/xceptance/xlt/engine/metrics/otel/OpenTelemetryConfiguration.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (h-arlt) [reply src/main/java/com/xceptance/xlt/engine/metrics/otel/OpenTelemetryConfiguration.java]: Done.

**Review by jowerner — APPROVED** [review]

**Pull request merged**